File: admins/src/main/java/com/github/admins/payload/OrderDetail.java
package com.github.admins.payload;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderDetail implements Serializable, Cloneable {

    private static final long serialVersionUID = -8001976834441332571L;

    @JsonProperty(value = "id")
    private Long id;

    @NotNull
    @JsonProperty(value = "customer")
    private Customer customer;

    @NotEmpty
    @JsonProperty(value = "productIds")
    private List<Long> productIds;

    @NotNull
    @JsonProperty(value = "amount")
    private BigDecimal amount;

    @NotNull
    @JsonProperty(value = "userId")
    private Long userId;

    @NotNull
    @JsonProperty(value = "status")
    private OrderStatus status;

    public OrderDetail(Long id,
                       List<Long> productIds,
                       Customer customer,
                       BigDecimal amount,
                       OrderStatus status) {
        this.id = id;
        this.productIds = productIds;
        this.customer = customer;
        this.amount = amount;
        this.status = status;
    }

}

File: server-mocks/src/main/kotlin/com/github/server/mocks/payload/OrderDetail.kt
package com.github.server.mocks.payload

import java.math.BigDecimal

class OrderDetail(
        val id: Long? = null,
        val customer: Customer? = null,
        val productIds: List<Long>? = null,
        val amount: BigDecimal? = null,
        val userId: Long? = null,
        val status: OrderStatus? = null
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as OrderDetail

        if (id != other.id) return false
        if (customer != other.customer) return false
        if (productIds != other.productIds) return false
        if (amount != other.amount) return false
        if (userId != other.userId) return false
        if (status != other.status) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (customer?.hashCode() ?: 0)
        result = 31 * result + (productIds?.hashCode() ?: 0)
        result = 31 * result + (amount?.hashCode() ?: 0)
        result = 31 * result + (userId?.hashCode() ?: 0)
        result = 31 * result + (status?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "OrderDetail(id=$id, customer=$customer, productIds=$productIds, amount=$amount, userId=$userId, status=$status)"
    }
}